doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,379 @@ permalink: /:path/
+This will make possible to tightly integrate your application's bulid system with Kaa SDK and use already provided ESP8266 toolchain file.
+
+#### CMakeLists.txt
+
+First, let's tell CMake minimum version required and project name.
+```CMake
+cmake_minimum_required(VERSION 3.0.2)
+
+project(kaa_demo C)
+```
+
+We also add Kaa SDK subdirectory, so that we can build it together with application.
+
+```CMake
+add_subdirectory(libs/kaa)
+```
+
+Next, let's create a static library with demo source files.
+
+```CMake
+# Add source files
+add_library(kaa_demo_s STATIC user/user_main.c driver/uart.c src/kaa_demo.c)
+```
+
+We also enable c99 standard:
+
+```CMake
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+```
+
+This part adds required include directories to compiler search path.
+
+```CMake
+if(NOT DEFINED ESP_RTOS_SDK)
+    set(ESP_RTOS_SDK /opt/Espressif/esp-rtos-sdk)
+endif()
+
+# specify include directories
+target_include_directories(kaa_demo_s PUBLIC driver)
+target_include_directories(kaa_demo_s PUBLIC .)
+target_include_directories(kaa_demo_s PUBLIC
+                           ${ESP_RTOS_SDK}/extra_include
+                           ${ESP_RTOS_SDK}/include
+                           ${ESP_RTOS_SDK}/include/lwip
+                           ${ESP_RTOS_SDK}/include/lwip/ipv4
+                           ${ESP_RTOS_SDK}/include/lwip/ipv6
+                           ${ESP_RTOS_SDK}/include/espressif/
+                           )
+```
+
+We also need to use custom linker script.
+Below, two linker scripts, required for ESP8266 applications, are moved to binary directory so that linker can locate them.
+
+```CMake
+file(COPY
+    ${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.rom.addr.v6.ld
+    ${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.app.v6.ld
+    DESTINATION
+    ${CMAKE_BINARY_DIR})
+```
+
+Next, we should tell CMake what libraries we would like to link with.
+Here the required libraries from ESP8266 RTOS SDK, linker script and Kaa SDK are specified.
+
+```CMake
+exec_program(xtensa-lx106-elf-gcc .
+            ARGS -print-libgcc-file-name
+            OUTPUT_VARIABLE ESP8266_LIBGCC
+            )
+
+target_link_libraries(kaa_demo_s PUBLIC
+                      kaac
+                      ${ESP_RTOS_SDK}/lib/libfreertos.a
+                      ${ESP_RTOS_SDK}/lib/libhal.a
+                      ${ESP_RTOS_SDK}/lib/libpp.a
+                      ${ESP_RTOS_SDK}/lib/libphy.a
+                      ${ESP_RTOS_SDK}/lib/libnet80211.a
+                      ${ESP_RTOS_SDK}/lib/libwpa.a
+                      ${ESP_RTOS_SDK}/lib/liblwip.a
+                      ${ESP_RTOS_SDK}/lib/libmain.a
+                      ${ESP_RTOS_SDK}/lib/libssl.a
+                      ${ESP_RTOS_SDK}/lib/libhal.a
+                      ${ESP8266_LIBGCC}
+                      -T${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.app.v6.ld
```

Comment:
`${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.app.v6.ld` is already copied to the `${CMAKE_BINARY_DIR}` 
To keep consistency, use one that located in binary dir